latch/ldata/path.py

```diff
@@ -0,0 +1,272 @@
+import os
+import re
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import Generator, Optional, Type, Union
+
+import gql
+from flytekit import (
+    Blob,
+    BlobMetadata,
+    BlobType,
+    FlyteContext,
+    Literal,
+    LiteralType,
+    Scalar,
+)
+from flytekit.extend import TypeEngine, TypeTransformer
+from latch_sdk_gql.execute import execute
+
+from latch.ldata.transfer.node import LDataNodeType
+from latch.ldata.transfer.progress import Progress
+from latch_cli.utils import urljoins
+
+from .transfer.download import _download
+from .transfer.remote_copy import _remote_copy
+from .transfer.upload import _upload
+
+node_id_regex = re.compile(r"^latch://(?P<id>[0-9]+)\.node$")
+
+
+dir_types = {
+    LDataNodeType.dir,
+    LDataNodeType.account_root,
+    LDataNodeType.mount,
+}
+
+
+@dataclass
+class _Cache:
+    """Internal cache class to organize information for a `LPath`."""
+
+    path: Optional[str] = None
+    node_id: Optional[str] = None
+    name: Optional[str] = None
+    type: Optional[LDataNodeType] = None
+    size: Optional[int] = None
+    content_type: Optional[str] = None
+
+
+@dataclass
+class LPath:
+
+    _cache: _Cache = field(
+        repr=False,
+        hash=False,
+        compare=False,
+    )
+
+    path: str
+
+    def __init__(self, path: Union[str, "LPath"]):
+        if isinstance(path, LPath):
+            path = path.path
+        if not path.startswith("latch://"):
+            raise ValueError(f"Invalid LPath: {path} is not a Latch path")
+        self.path = path
+        self._cache = _Cache()
+        self._download_idx = 0
+
+    def load(self) -> None:
```

Comment:
```suggestion
    def load_metadata(self) -> None:
```

or something similar - could be confusing to people who might think that `load == download` lol